src/index.js

```diff
@@ -400,35 +400,37 @@ export default class Carousel extends React.Component {
   }
   // eslint-disable-next-line complexity
   handleKeyPress(e) {
```

Comment:
instead of checking for the `disableKeyboardControls` prop inside of this function, you can check for it before the method is even called. That way, the method isn't called if `disableKeyboardControls` is false

Reply:
on lines 695 and 720

Reply:
@sarmeyer I would be worry that this could cause a potential bug.  If the user changes `disableKeyboardControls` at any point it may not get caught right away, depending on how the user changes the prop. If we move the check on the mount we will also need to add / remove / update the event listener during the `componentWillReceiveProps`. Alternatively, we won't be listening to changes to that prop. Since we are trying to simplify `componentWillReceiveProps` as part of the refactor I would consider leave the check in the `handleKeypress` event.

Reply:
Yea, I understand. We can leave the check where it is for now